src/main/java/oracle/kubernetes/operator/rest/model/UpgradeApplicationModel.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator.rest.model;
+
+/**
+ * UpgradeApplicationModel describes the input data for each application to the WebLogic application upgrade operation.
+ */
+public class UpgradeApplicationModel extends BaseModel {
+
+  private String applicationName;
+  private String patchedLocation;
+  private String backupLocation;
+
+  /**
+   * Get the name of the application to be upgraded..
+   * @return the name of the application..
+   */
+  public String getApplicationName() {
+    return applicationName;
+  }
+
+  /**
+   * Get the location of the new version of application.
+   * @return
+   */
+  public String getPatchedLocation() {
+    return patchedLocation;
+  }
+
+  /**
+   * Get the location the current version of application to be backed up.
+   * @return
+   */
+  public String getBackupLocation() {
```

Comment:
Do we default this?  Are we going to provide a roll-back operation?

Reply:
I think eventually we should and it's part of rolling upgrade. But maybe for the first stage, we can just focus on the upgrade operation. What do you think?